Create code-review-dance-activity.md

blakeLarkin opened the pull request:
This is my new version of the code review activity, based on the existing Github Tutorial.

blakeLarkin committed:
Create code-review-dance-activity.md

This is my new version of the code review activity, based on the existing Github Tutorial.

ykharitonova commented on code-review-dance-activity.md:
```diff
@@ -0,0 +1,112 @@
+- You know how to open a terminal and generally work from the command-line.
+- You know enough of vi or vim to edit, move around in, save, and quit files.
+- You have completed the other Github tutorial and have a basic understanding of it's features
+- You have a basic repository, akin to the one created in the other tutorial, with mandatory code reviews
```
what does it mean: "with mandatory code reviews"?

ykharitonova replied:
Is it what you explained in Step 1? If they do it there, then it's not a prereq?

ykharitonova commented on code-review-dance-activity.md:
```diff
@@ -0,0 +1,112 @@
+
+Roles:
+
+-   Developer A: Set's up required reviews and reviews Developer B's pull request (PR)
```
Set's up --> Sets up

ykharitonova commented on code-review-dance-activity.md:
```diff
@@ -0,0 +1,112 @@
+
+-   Developer A: Set's up required reviews and reviews Developer B's pull request (PR)
+
+-   Developer B: Create's an erroneous addition and attempts to create a pull request
```
no need for an apostrophe

ykharitonova commented on code-review-dance-activity.md:
```diff
@@ -0,0 +1,112 @@
+
+If not already the case, make code reviews required for pull requests of important branches (Step 1 below).
+This helps prevent errorneous code (2) from slipping into your code base from a Pull Request (3).
+Performing these code reviews (4) foster learning from your development team and sharing your own expertise.
```
fosters

ykharitonova commented on code-review-dance-activity.md:
```diff
@@ -0,0 +1,112 @@
+    * A TODO comment like this, similar to test statements, should never make it onto your master branch and should be removed before submitting a pull request
+  * Create a for loop that loops from 0 to n times
+  * Create an inner for loop that loops from 0 to 3 times
+  * Finally, inside the inner for loop print “Hi” with “puts “Hi””
```
use code marks, i.e. `backticks`, for code-related lines

ykharitonova commented on code-review-dance-activity.md:
```diff
@@ -0,0 +1,112 @@
+*On Github*
+### (4) Review the new PR (Developer A)
+
+  Navigate to the "Pull Requests" tab and open the new PR. Click on the “Files Changed” tab to see what the pull request covers.
```
the "Pull Requests" tab on GitHub

ykharitonova commented on code-review-dance-activity.md:
```diff
@@ -0,0 +1,112 @@
+*On Github and locally*
+#### (5) Respond to comments (Developer B)
+
+  Check your pull request on Github and read over the comments Developer A has 
```
Check your "Pull Requests" tab on GitHub

ykharitonova commented on code-review-dance-activity.md:
```diff
@@ -0,0 +1,112 @@
+  **You’ve approved the changes, but don’t merge the branch! Let Developer B merge the branch to make sure the git blame log matches who worked on the branch**
+
+### (7) Merge the PR! (Developer B)
+  Refresh your pull request. Since Developer A has approved it, the merge button should now be clickable. Merge it!
```
Isn't this dependent on the configuration of the repo? I.e., is it enough to just be the collaborator on the repo to merge the commit in?

blakeLarkin replied:
Yes, I changed the instructions earlier to allow collaborators to approve PRs (not only owners). I believe if they are in the same groups as the first assignment it should work.

blakeLarkin commented:
Need to explain how to make a PR and make it explicit to use the repo from the other exercise

blakeLarkin commented on code-review-dance-activity.md:
```diff
@@ -0,0 +1,112 @@
+*On GitHub*
+### (1) Set up required code reviews (if necessary) (Developer A)
+
+  Go to the project's "Settings" tab and from the left sidebar select "Branches". Under the "Protected Branches" select the master branch to protect it from deletion and pushes. Require changes to master be made through pull-requests, and require each pull-request be reviewed by a maintainer (i.e., select "Require review from Code Owners").
```
Clarify what is meant by "project's 'Settings'"

blakeLarkin commented on code-review-dance-activity.md:
```diff
@@ -0,0 +1,112 @@
+  Save the file and add your changes. Push your changes and create 
+  a pull request comparing your branch to master. Give your pull 
+  request a good name, and to notify Developer A that you’ve made a 
+  PR by including “@[Developer A’s github handle]” in the description
```
This is where I may need to describe making PRs

blakeLarkin commented on code-review-dance-activity.md:
```diff
@@ -0,0 +1,112 @@
+### (4) Review the new PR (Developer A)
+
+  Navigate to the "Pull Requests" tab and open the new PR. Click on the “Files Changed” tab to see what the pull request covers.
+  Now comment on the problems in the code, including:
```
Say how to leave comments

blakeLarkin replied:
explain difference between single comments and a review

blakeLarkin replied:
and if doing a review make sure that you don't approve it if there are issues

blakeLarkin replied:
Infact, select "Request Changes"

blakeLarkin commented on code-review-dance-activity.md:
```diff
@@ -0,0 +1,112 @@
+
+  Check your pull request on Github and read over the comments Developer A has 
+  provided. Fix the issues they pointed out and answer any questions they left.
+  Git add and push your changes and verify that your changes have been updated on the PR.
```
leave good commit messages

blakeLarkin commented on code-review-dance-activity.md:
```diff
@@ -0,0 +1,112 @@
+well understood by the development team and up to par with the rest of the repository. By
+following this format your development team will all be on the same page about the status of
+the repository and every gets a chance to learn from one another.
+
```
Talk about deleting branches

blakeLarkin replied:
I added this

ykharitonova reviewed (changes requested):
Looks like my comments are currently "pending"; let's see if "request changes" puts them through...

blakeLarkin committed:
Addressed comments from YK and grutor tryout

I've updated this to address changes suggested by YK and notes that arose during trying it out with another grutor.

blakeLarkin committed:
Added warning about finishing reviews

Added a quick note that explains that reviews must be finished if you want anyone to see your comments

blakeLarkin committed:
Clarify final approval process

Small change to part 6

azwildcat reviewed (approved):
Let's merge this activity into the main repo.

blakeLarkin committed:
Update to make a new repo

Removed specific dependency on other activity.

blakeLarkin committed:
Added important option
